bundles/org.openhab.binding.wemo/src/main/java/org/openhab/binding/wemo/internal/handler/WemoCrockpotHandler.java

```diff
@@ -0,0 +1,323 @@
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("WeMoCrockpotHandler disposed.");
+
+        removeSubscription();
+
+        if (refreshJob != null && !refreshJob.isCancelled()) {
+            refreshJob.cancel(true);
+            refreshJob = null;
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.trace("Command '{}' received for channel '{}'", command, channelUID);
+        String mode = "0";
+        String time = null;
+
+        if (command instanceof RefreshType) {
+            try {
+                updateWemoState();
+            } catch (Exception e) {
+                logger.debug("Exception during poll", e);
+            }
+        } else if (channelUID.getId().equals(CHANNEL_COOKMODE)) {
+            String commandString = command.toString();
+            switch (commandString) {
+                case "OFF":
+                    mode = "0";
+                    time = "0";
+                    break;
+                case "WARM":
+                    mode = "50";
+                    break;
+                case "LOW":
+                    mode = "51";
+                    break;
+                case "HIGH":
+                    mode = "52";
+                    break;
+            }
+            try {
+                String soapHeader = "\"urn:Belkin:service:basicevent:1#SetBinaryState\"";
+                String content = "<?xml version=\"1.0\"?>"
+                        + "<s:Envelope xmlns:s=\"http://schemas.xmlsoap.org/soap/envelope/\" s:encodingStyle=\"http://schemas.xmlsoap.org/soap/encoding/\">"
+                        + "<s:Body>" + "<u:SetCrockpotState xmlns:u=\"urn:Belkin:service:basicevent:1\">" + "<mode>"
+                        + mode + "</mode>" + "<time>" + time + "</time>" + "</u:SetCrockpotState>" + "</s:Body>"
+                        + "</s:Envelope>";
+                String wemoURL = getWemoURL("basicevent");
+
+                if (wemoURL != null) {
+                    wemoHttpCaller.executeCall(wemoURL, soapHeader, content);
+                }
+            } catch (Exception e) {
+                logger.error("Failed to send command '{}' for device '{}': {}", command, getThing().getUID(),
+                        e.getMessage());
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+            }
+            updateStatus(ThingStatus.ONLINE);
+        }
+    }
+
+    @Override
+    public void onServiceSubscribed(String service, boolean succeeded) {
+        logger.debug("WeMo {}: Subscription to service {} {}", getUDN(), service, succeeded ? "succeeded" : "failed");
+        subscriptionState.put(service, succeeded);
+    }
+
+    @Override
+    public void onValueReceived(String variable, String value, String service) {
+        logger.debug("Received pair '{}':'{}' (service '{}') for thing '{}'",
```

Comment:
This is really old fashioned, please use the normal var-args call

Reply:
Don‘t get your point, do you have an example?

Reply:
Just use the following instead of having the `new Object[]` 

```
logger.debug("Received pair '{}':'{}' (service '{}') for thing '{}'", variable, value, service, this.getThing().getUID());

Reply:
Done